### DISCUSSION TITLE
How to setup a propper CCButtonLatch Array?

### DISCUSSION THREAD

**Discussion by sincos5** [tttapa/Control-Surface#534] [Q&A] [answered]
Hey! im trying to set up a working Button Latch array with Channel option, but i cant. Can anyone help?
`CCButtonLatched buttons [] = {
  {2, MIDI_CC::General_Purpose_Controller_1},
  {3, MIDI_CC::General_Purpose_Controller_2},
};`

this is my try, but when i put the channel info an error pops up.

`CCButtonLatched buttons [] = {
  {2, MIDI_CC::General_Purpose_Controller_1,CHANNEL_2},
  {3, MIDI_CC::General_Purpose_Controller_2,CHANNEL_2},
};`

also, anyone knows where to get info about this?

**Comment by tttapa** [comment, marked as answer]
The MIDI address is a single argument, you have to group the controller number and the channel using a pair of curly braces, please see §4 of https://tttapa.github.io/Control-Surface-doc/Doxygen/d5/d7d/md_pages_Getting-Started.html.

**Reply by sincos5** [reply]
it would be good to have an example named ARRAYS that comes as a default example
thanks as always!
